cluster_test.go

```diff
@@ -24,54 +24,107 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestUrl(t *testing.T) {
-	cluster, _ := newCluster(http.DefaultClient, fakeMarathonURL)
-	assert.Equal(t, cluster.URL(), fakeMarathonURL)
-}
-
 func TestSize(t *testing.T) {
 	cluster, _ := newCluster(http.DefaultClient, fakeMarathonURL)
-	assert.Equal(t, cluster.Size(), 3)
+	assert.Equal(t, cluster.size(), 3)
 }
 
 func TestActive(t *testing.T) {
 	cluster, _ := newCluster(http.DefaultClient, fakeMarathonURL)
-	assert.Equal(t, len(cluster.Active()), 3)
+	assert.Equal(t, len(cluster.activeMembers()), 3)
 }
 
 func TestNonActive(t *testing.T) {
 	cluster, _ := newCluster(http.DefaultClient, fakeMarathonURL)
-	assert.Equal(t, len(cluster.NonActive()), 0)
+	assert.Equal(t, len(cluster.nonActiveMembers()), 0)
 }
 
 func TestGetMember(t *testing.T) {
 	cluster, _ := newCluster(http.DefaultClient, fakeMarathonURL)
-	member, err := cluster.GetMember()
+	member, err := cluster.getMember()
 	assert.NoError(t, err)
 	assert.Equal(t, member, "http://127.0.0.1:3000")
 }
 
 func TestGetMemberWithPath(t *testing.T) {
 	cluster, _ := newCluster(http.DefaultClient, fakeMarathonURLWithPath)
-	member, err := cluster.GetMember()
+	member, err := cluster.getMember()
 	assert.NoError(t, err)
 	assert.Equal(t, member, "http://127.0.0.1:3000/path")
 }
 
 func TestMarkDown(t *testing.T) {
 	endpoint := newFakeMarathonEndpoint(t, nil)
 	defer endpoint.Close()
-
 	cluster, _ := newCluster(http.DefaultClient, endpoint.URL)
-	assert.Equal(t, len(cluster.Active()), 3)
-	cluster.MarkDown()
-	cluster.MarkDown()
-	assert.Equal(t, len(cluster.Active()), 1)
+	assert.Equal(t, len(cluster.activeMembers()), 3)
+
+	members := cluster.activeMembers()
+	cluster.markDown(members[0])
+	cluster.markDown(members[1])
+	assert.Equal(t, 1, len(cluster.activeMembers()))
```

Comment:
This check and `healthCheckNode` seem to race here. Not exactly sure how to solve this easily though.